crates/blockchain/payload.rs

```diff
@@ -96,7 +101,19 @@ pub fn build_payload(args: &BuildPayloadArgs, storage: &Store) -> Result<Block,
             ommers: Vec::new(),
             withdrawals: Some(args.withdrawals.clone()),
         },
-    })
+    };
+    // Apply withdrawals & call beacon root contract, and obtain the new state root
+    let spec_id = spec_id(storage, args.timestamp)?;
+    let mut evm_state = evm_state(storage.clone(), parent_block.number);
+    if args.beacon_root.is_some() && spec_id == SpecId::CANCUN {
+        beacon_root_contract_call(&mut evm_state, &payload.header, spec_id)?;
+    }
+    process_withdrawals(&mut evm_state, &args.withdrawals)?;
+    let account_updates = get_state_transitions(&mut evm_state);
+    payload.header.state_root = storage
+        .apply_account_updates(parent_block.number, &account_updates)?
```

Comment:
Similar to https://github.com/lambdaclass/ethereum_rust/issues/443 and maybe out of scope of this PR, but you should be able to do this even when `parent_block` is not canonical. So `apply_account_updates` should receive a block hash instead of a block number